src/lib/address_resolve/tests/TestAddressResolve_DefaultImpl.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <lib/address_resolve/AddressResolve_DefaultImpl.h>
+
+#include <lib/support/UnitTestRegistration.h>
+
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::AddressResolve;
+
+namespace {
+
+constexpr uint8_t kNumberOfAvailableSlots = CHIP_CONFIG_MDNS_RESOLVE_LOOKUP_RESULTS;
+
+const Transport::PeerAddress GetAddressWithLowScore(uint16_t port                 = CHIP_PORT,
+                                                    Inet::InterfaceId interfaceId = Inet::InterfaceId::Null())
+{
+    Inet::IPAddress ipAddress;
+    Inet::IPAddress::FromString("::/80", ipAddress);
+
+    return Transport::PeerAddress::UDP(ipAddress, port, interfaceId);
+}
+
+const Transport::PeerAddress GetAddressWithMediumScore(uint16_t port                 = CHIP_PORT,
+                                                       Inet::InterfaceId interfaceId = Inet::InterfaceId::Null())
+{
+    Inet::IPAddress ipAddress;
+    Inet::IPAddress::FromString("192.168.0.1", ipAddress);
```

Comment:
IPv4 addresses should not be considered in this test unless IPv4 support is enabled. Please remove IPv4 example.